src/realtime_helpers.cpp

```diff
@@ -180,9 +180,18 @@ std::pair<bool, std::string> set_thread_affinity(int pid, int core)
 #endif
 }
 
+std::pair<bool, std::string> set_thread_affinity(std::thread & thread, int core)
+{
+  if (!thread.joinable()) {
```

Comment:
Just out of interest: Why does a thread need to be joinable in order to set its affinity?

Reply:
Its just to kinda check if the thread is initialized or not. If not, there is no other way to check apart from that. 

If it is not joinable, then it's not active, so it doesn't make sense IMO